kubernetes/resource_kubernetes_pod.go

```diff
@@ -15,6 +15,11 @@ import (
 )
 
 func resourceKubernetesPod() *schema.Resource {
+	podSpecFields := podSpecFields(false, false, false)
+	// Setting this default to false prevents a perpetual diff caused by volume_mounts
+	// being mutated on the server side as Kubernetes automatically adds a mount
+	// for the service account token
+	podSpecFields["automount_service_account_token"].Default = false
```

Comment:
FWIW - I resolved this issue in my fork of the provider:
https://github.com/sl1pm4t/terraform-provider-kubernetes/commit/e0e953c94a66c7e6070a5db5c1c5475b3a6992e1#diff-7424630baa3c87d787d88925b5f81cac